### PR TITLE
fixes #736 avoid dropping exceptions in scanner

### LINKED ISSUE #736: Scanner throws exception from background thread.
While working on #718, an IT was iterating over a scanner that was created from a closed AccumuloClient.  This caused the following code to throw an exception from a background thread.

https://github.com/apache/accumulo/blob/rel/1.9.2/core/src/main/java/org/apache/accumulo/core/client/impl/ScannerIterator.java#L162

Since the exception was from a background thread, I had no idea what code in the IT was using the scanner.  I tried changing the code to wrap the exception from the background thread, but that broke other code looking for specific exceptions like IsolationException.

Need to do something to improve debugging while not breaking anything.

### PR DESCRIPTION
**Pull request by keith-turner**
In addition to cleaning up the exception handling I also switched from a background thread putting a batch of mutations on a queue to using a future.  Doing this made the code simpler making it easier to clean up the exception handling.

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/clientImpl/ScannerIterator.java [hunk 1]
```diff
@@ -134,55 +98,66 @@ public void run() {
...
-
-      Object obj = synchQ.take();
-
-      if (obj instanceof Exception) {
-        finished = true;
-        if (obj instanceof RuntimeException)
-          throw (RuntimeException) obj;
-        else
-          throw new RuntimeException((Exception) obj);
+    } catch (ExecutionException ee) {
```
Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/clientImpl/ScannerIterator.java]: The code in this catch block is what fixes #736.  A lot of the other changes are the refactoring to use future to execute code in the background.

File: core/src/main/java/org/apache/accumulo/core/clientImpl/ScannerIterator.java [hunk 2]
```diff
@@ -134,55 +98,66 @@ public void run() {
...
+      if (ee.getCause() instanceof IsolationException)
+        throw new IsolationException(ee);
+      if (ee.getCause() instanceof TableDeletedException) {
+        TableDeletedException cause = (TableDeletedException) ee.getCause();
+        throw new TableDeletedException(cause.getTableId(), cause);
```
[the remark this answers is not in the document]
Comment (keith-turner) [reply core/src/main/java/org/apache/accumulo/core/clientImpl/ScannerIterator.java]: Doing that loses the code path the the current/foreground thread took to get to this point.  The cause is an exception that occurred in a background thread.

[the remark this answers is not in the document]
Comment (ctubbsii) [reply core/src/main/java/org/apache/accumulo/core/clientImpl/ScannerIterator.java]: Gotcha.

File: core/src/main/java/org/apache/accumulo/core/clientImpl/ScannerIterator.java (hunk 1 above)
Comment (milleruntime) [reply core/src/main/java/org/apache/accumulo/core/clientImpl/ScannerIterator.java]: Could create handleExecutionException() to make this a bit easier to read.

**Pull request merged**